services/storage/src/simcore_service_storage/handlers.py

```diff
@@ -78,6 +78,32 @@
         }
 
 
+async def get_datasets_metadata(request: web.Request):
+    log.debug("GET METADATA DATASETS %s %s",request.path, request.url)
+
+    params, query, body = await extract_and_validate(request)
+
+    assert params, "params %s" % params
+    assert query, "query %s" % query
+    assert not body, "body %s" % body
+
+    assert params["location_id"]
+    assert query["user_id"]
+
+    location_id = params["location_id"]
+    user_id = query["user_id"]
+
+    dsm = await _prepare_storage_manager(params, query, request)
+
+    location = dsm.location_from_id(location_id)
+    # To implement
+    data = await dsm.list_datasets(user_id, location)
+
+    return {
```

Comment:
FYI: you are not required to provide the envelope. The [envelope middleware](https://github.com/ITISFoundation/osparc-simcore/blob/master/packages/service-library/src/servicelib/rest_middlewares.py#L139) will take care of that. You can just return the data or raise an exception if there is an error